edx_rest_framework_extensions/auth/jwt/authentication.py

```diff
@@ -371,6 +332,34 @@ def _is_jwt_cookie_and_session_user_mismatch(self, request, jwt_user_id):
 
         return True
 
+    def _get_and_monitor_jwt_cookie_lms_user_id(self, request):
+        """
+        Returns the LMS user id from the JWT cookie, or None if not found
+
+        Notes:
+        - Must only be called in the case of a JWT cookie.
+        - Also provides monitoring details for mismatches.
+        """
+        try:
+            cookie_token = JSONWebTokenAuthentication.get_token_from_cookies(request.COOKIES)
+            invalid_decoded_jwt = unsafe_jwt_decode_handler(cookie_token)
```

Comment:
**Author note**:
1. Reviewing by commit may be simpler, because this method was moved in a separate commit.
2. We will now be decoding all JWT cookies, which hopefully won't be a performance issue since we aren't even verifying the signature or anything.

Review can mark as resolved once read.